src/design-system/components/checkbox/CheckBox.tsx

```diff
@@ -0,0 +1,78 @@
+import Checkbox, { CheckboxProps } from '@mui/material/Checkbox'
+import { FC, ReactNode, useState } from 'react'
+import Loader from '~/components/loader/Loader'
+
+import './CheckBox.scss'
+
+interface S2SCheckBoxProps extends Omit<CheckboxProps, 'size'> {
+  variant: 'check' | 'middle'
+  label: ReactNode
+  labelPosition?: 'top' | 'bottom' | 'end'
+  color?: 'primary' | 'secondary' | 'error' | 'success'
+  size?: 'sm' | 'md' | 'lg'
+  loading?: boolean
+}
+
+const CheckBox: FC<S2SCheckBoxProps> = ({
+  color = 'primary',
+  disabled = false,
+  label,
+  labelPosition = 'end',
+  loading = false,
+  variant = 'check',
+  size = 'md',
+  ...props
+}) => {
+  const [checked, setChecked] = useState<boolean>(false)
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChecked(event.target.checked)
+  }
+
+  const blockModifiers = [
```

Comment:
Use `cn` function here, please
Look [here](https://github.com/ita-social-projects/SpaceToStudy-Client/blob/develop/src/design-system/components/button/Button.tsx#L67) for example